service/storage_service.cc

```diff
@@ -1967,8 +1967,16 @@ class topology_coordinator : public endpoint_lifecycle_subscriber {
             if (holder && holder->failed()) {
                 holder->ignore_ready_future(); // Prevent warnings in logs
             }
-            holder = futurize_invoke(action)
-                        .finally([this, g = _async_gate.hold(), gid, name] () noexcept {
+            holder = futurize_invoke(action).then_wrapped([this, gid, name] (future<> f) {
+                if (f.failed()) {
+                    auto ep = f.get_exception();
+                    slogger.error("raft topology: {} for tablet {} failed: {}", name, gid, ep);
+                    return seastar::sleep_abortable(std::chrono::seconds(1), _as).then([ep] () mutable {
+                        std::rethrow_exception(ep);
+                    });
+                }
```

Comment:
Overall it would be better to have a fiber own the work, rather than managing random futures. But that's outside the scope here.